src/hooks/usePersistentGlobalState.ts

```diff
@@ -31,3 +32,31 @@ export const createPersistentGlobalStateHook = <T extends object>(
 
     return () => [container.useGlobalState(key)[0], setGlobalState];
 };
+
+export const usePersistentGlobalState = <T extends object>(
```

Comment:
This one is actually quite clever. I was a bit sceptical at first of using useSWR for this, but if there aren't any drawbacks it will allow us to get rid of a dependency.